File: Interviews/LeetCode/Daily-Problems/2022/October/Day-07/732.cpp
class MyCalendarThree {
public:
    map<int, int> interval;
    MyCalendarThree() {
        interval.clear();
    }

    int book(int start, int end) {
        interval[start] += 1;
        interval[end] -= 1;

        int cur = 0, ans = 0;
        for (auto i : interval) {
            cur += i.second;
            ans = max(ans, cur);
        }

        return ans;
    }
};

File: Interviews/LeetCode/Daily-Problems/2022/October/Day-07/732.py
class MyCalendarThree:
    def __init__(self):
        self.interval = {}

    def book(self, start: int, end: int) -> int:
        self.interval[start] = self.interval.get(start, 0) + 1
        self.interval[end] = self.interval.get(end, 0) - 1
            
        cur, ans = 0, 0
        for i in sorted(self.interval):
            cur += self.interval[i]
            ans = max(ans, cur)
        
        return ans